app/src/org/commcare/utils/StorageUtils.java

```diff
@@ -49,74 +63,47 @@ public static int getNumIncompleteForms() {
                 new String[]{FormRecord.STATUS_INCOMPLETE, currentAppId}).size();
     }
 
-    public static FormRecord[] getUnsentRecords(SqlStorage<FormRecord> storage) {
+    public static FormRecord[] getUnsentRecordsForCurrentApp(SqlStorage<FormRecord> storage,
```

Comment:
In the past we were only sorting the records conditionally because of how expensive it was to parse out all of the date strings. Given that the sort data is now super simple and included in the record pull already, I'd be super supportive of just always sorting rather than having multiple ways to pull the list.